core/src/main/java/jenkins/install/InstallState.java

```diff
@@ -24,64 +24,75 @@
 package jenkins.install;
 
 import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
 
-import org.kohsuke.accmod.Restricted;
-import org.kohsuke.accmod.restrictions.NoExternalUse;
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
 
 /**
  * Jenkins install state.
  *
  * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
  */
-@Restricted(NoExternalUse.class)
-public enum InstallState {
+public class InstallState implements ExtensionPoint {
```

Comment:
@oleg-nenashev It was `@Restricted`, so I don't understand what you're saying. This could as well be a different type, with larger changes to core to refer to the new type, and we could delete the old one.